### ISSUE TITLE
Linux Compile Error - Alloca

### ISSUE THREAD

**Issue by GregSlazinski** [NVIDIAGameWorks/PhysX#387]
```
In file included from /home/greg/Desktop/Esenthel/ThirdPartyLibs/PhysX/physx/source/pvd/src/PxProfileEventImpl.cpp:29:
In file included from /home/greg/Desktop/Esenthel/ThirdPartyLibs/PhysX/physx/source/pvd/src/PxProfileZoneImpl.h:38:
/home/greg/Desktop/Esenthel/ThirdPartyLibs/PhysX/physx/source/pvd/src/PxProfileEventBufferAtomic.h:293:5: error: use of function '__builtin_alloca' is discouraged; there is no way to check for failure but failure may still occur, resulting in a possibly exploitable security vulnerability [-Werror,-Walloca]
                                PX_ALLOCA(tempBuffer, uint8_t, sizeToWrite);
                                ^
/home/greg/Desktop/Esenthel/ThirdPartyLibs/PhysX/physx/source/foundation/include/PsAlloca.h:73:43: note: expanded from macro 'PX_ALLOCA'
                        var.mPointer = reinterpret_cast<type*>(PxAlloca(size));                                                    \
                                                               ^
/home/greg/Desktop/Esenthel/ThirdPartyLibs/PhysX/physx/source/foundation/include/PsAllocator.h:101:22: note: expanded from macro 'PxAlloca'
        #define PxAlloca(x) alloca(x)
                            ^
/usr/include/alloca.h:35:23: note: expanded from macro 'alloca'
# define alloca(size)   __builtin_alloca (size)
                        ^
1 error generated.

```
clang version 10.0.0-4ubuntu1 
Target: x86_64-pc-linux-gnu

**Comment by heyuanYao-pku** [comment]
I met the same problem, is there any solution? thx!

**Comment by GregSlazinski** [comment]
I've integrated all known fixes and workarounds over to my game engine's version:
https://github.com/Esenthel/EsenthelEngine/tree/master/ThirdPartyLibs/PhysX

**Comment by Hoikas** [comment]
PhysX compiles with `-Werror` by default. You can remove it with this patch or by using the physx port in vcpkg.

```diff
diff --git a/physx/source/compiler/cmake/linux/CMakeLists.txt b/physx/source/compiler/cmake/linux/CMakeLists.txt
index 13acf52e..b709b9f9 100644
--- a/physx/source/compiler/cmake/linux/CMakeLists.txt
+++ b/physx/source/compiler/cmake/linux/CMakeLists.txt
@@ -28,8 +28,8 @@
 STRING(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_LOWERCASE)
 
 #TODO: Fix warnings
-SET(CLANG_WARNINGS "-ferror-limit=0 -Wall -Wextra -Werror -Wstrict-aliasing=2 -Weverything -Wno-documentation-deprecated-sync -Wno-documentation-unknown-command -Wno-gnu-anonymous-struct -Wno-undef -Wno-unused-function -Wno-nested-anon-types -Wno-float-equal -Wno-padded -Wno-weak-vtables -Wno-cast-align -Wno-conversion -Wno-missing-noreturn -Wno-missing-variable-declarations -Wno-shift-sign-overflow -Wno-covered-switch-default -Wno-exit-time-destructors -Wno-global-constructors -Wno-missing-prototypes -Wno-unreachable-code -Wno-unused-macros -Wno-unused-member-function -Wno-used-but-marked-unused -Wno-weak-template-vtables -Wno-deprecated -Wno-non-virtual-dtor -Wno-invalid-noreturn -Wno-return-type-c-linkage -Wno-reserved-id-macro -Wno-c++98-compat-pedantic -Wno-unused-local-typedef -Wno-old-style-cast -Wno-newline-eof -Wno-unused-private-field -Wno-format-nonliteral -Wno-implicit-fallthrough -Wno-undefined-reinterpret-cast -Wno-disabled-macro-expansion -Wno-zero-as-null-pointer-constant -Wno-shadow -Wno-unknown-warning-option -Wno-atomic-implicit-seq-cst -Wno-extra-semi-stmt")
-SET(GCC_WARNINGS "-Wall -Werror -Wno-invalid-offsetof -Wno-uninitialized")
+SET(CLANG_WARNINGS "-ferror-limit=0 -Wall -Wextra -Wstrict-aliasing=2 -Weverything -Wno-documentation-deprecated-sync -Wno-documentation-unknown-command -Wno-gnu-anonymous-struct -Wno-undef -Wno-unused-function -Wno-nested-anon-types -Wno-float-equal -Wno-padded -Wno-weak-vtables -Wno-cast-align -Wno-conversion -Wno-missing-noreturn -Wno-missing-variable-declarations -Wno-shift-sign-overflow -Wno-covered-switch-default -Wno-exit-time-destructors -Wno-global-constructors -Wno-missing-prototypes -Wno-unreachable-code -Wno-unused-macros -Wno-unused-member-function -Wno-used-but-marked-unused -Wno-weak-template-vtables -Wno-deprecated -Wno-non-virtual-dtor -Wno-invalid-noreturn -Wno-return-type-c-linkage -Wno-reserved-id-macro -Wno-c++98-compat-pedantic -Wno-unused-local-typedef -Wno-old-style-cast -Wno-newline-eof -Wno-unused-private-field -Wno-format-nonliteral -Wno-implicit-fallthrough -Wno-undefined-reinterpret-cast -Wno-disabled-macro-expansion -Wno-zero-as-null-pointer-constant -Wno-shadow -Wno-unknown-warning-option -Wno-atomic-implicit-seq-cst -Wno-extra-semi-stmt")
+SET(GCC_WARNINGS "-Wall -Wno-invalid-offsetof -Wno-uninitialized")
 
 IF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
 	# using Clang	  
```

**Comment by heyuanYao-pku** [comment]
@Hoikas thanks a lot, it works well!

**Comment by ayoub-belarbi** [comment]
Thank you for reporting this issue! We just released PhysX 4.1.2 that fixes this issue (along with other fixes). Make sure to check it out :) 
Your contribution is greatly appreciated!
